lab_6_pipeline/pipeline.py

```diff
@@ -18,21 +33,47 @@ def __init__(self, path_to_raw_txt_data: Path):
         """
         Initializes CorpusManager
         """
+        self.path_to_raw_txt_data = path_to_raw_txt_data
+        self._validate_dataset()
+        self._storage = {}
+        self._scan_dataset()
 
     def _validate_dataset(self) -> None:
         """
         Validates folder with assets
         """
+        if not self.path_to_raw_txt_data.exists():
+            raise FileNotFoundError
+
+        if not self.path_to_raw_txt_data.is_dir():
+            raise NotADirectoryError
+
+        if not any(self.path_to_raw_txt_data.iterdir()):
+            raise EmptyDirectoryError
+
+        raw_files = [text for text in self.path_to_raw_txt_data.glob(r'*_raw.txt')]
+
+        list_existing = sorted(int(re.search(r'\d+', file.stem)[0]) for file in raw_files)
+        list_ideal = list(range(1, len(raw_files) + 1))
+        if list_existing != list_ideal:
```

Comment:
where is check for numbering of meta files?

Reply:
it wasn't needed for mark 4 but now it's done